articles/meta-learning/index.md

```diff
@@ -0,0 +1,101 @@
+
+**Grid Search.** This method makes use of manually predetermined hyperparameters. The group of predetermined parameters is searched for the best performing one. Grid search involves the trying of all possible combinations of hyperparameter values. The model then decides the best-suited hyperparameter value. However, this method is referred to as traditional since it is very time consuming and inefficient.
+
+**Random Search.** Grid search is an exhaustive method. It involves the tying of all possible combinations of values. The random search method replaces this exhaustive process with a random search. The model makes random combinations and attempts to fit the dataset to test for accuracy. Since the search is random, there is a possibility that the model may miss a few potentially optimal combinations. On the upside, it uses much less time compared to grid search and often gives ideal solutions. Random search can outperform grid search. This is under the condition that a few hyperparameters are required to optimize the algorithm.
+
+We shall cover these two and more methods of optimization in a different article. But for now, to learn more about grid search and random search, check out this [conceptual guide](https://medium.com/@jackstalfort/hyperparameter-tuning-using-grid-search-and-random-search-f8750a464b35) to hyperparameter tuning.
+
+#### Few-Shot Meta-Learning
+
+Deep learning algorithms are great at carrying out one task by using a sizeable dataset. Even so, it is desirable to be able to train a neural network to learn multiple tasks using a handful of data examples per task. Few-shot meta-learning algorithms help us fulfill this desire.
+
+The purpose of few-shot meta-learning is to train a model that can rapidly adapt to a new task. This is to be achieved using a handful of data points and iterations in training. A meta-learning stage is used to train a model on a given number of tasks. The expectation is that the trained model will quickly adapt to new tasks with a few trials or training examples. Entire tasks are taken as training examples in meta-learning.
+
+An example of few-shot meta-learning is the use of [memory-augmented neural networks](http://proceedings.mlr.press/v48/santoro16.pdf).
+
+[Gradient descent](https://ml-cheatsheet.readthedocs.io/en/latest/gradient_descent.html#:~:text=Gradient%20Descent.%20Gradient%20descent%20is%20an%20optimization%20algorithm,descent%20to%20update%20the%20parameters%20of%20our%20model.) is an optimization algorithm by nature. It minimizes a given function by moving towards the direction of the steepest descent iteratively. Gradient descent is used to update the parameters of a model. Traditional gradient descent networks need lots of data to learn. The training process is extensive and iterative. The models have to learn their parameters again to quickly add new information when exposed to new data. This is a very inefficient process.
```

Comment:
I don't think GD is a nature-based optimization algorithm. Ant-colonisation, PSO, etc are nature inspired.